python/pyarrow/array.pxi

```diff
@@ -1399,6 +1399,29 @@ cdef class Array(_PandasConvertible):
         """
         return _pc().index(self, value, start, end, memory_pool=memory_pool)
 
+    def sort(self, order="ascending", **options):
+        """
+        Sort the Array
+
+        Parameters
+        ----------
+        order : str, default "ascending"
+            Which order to sort values in.
+            Accepted values are "ascending", "descending".
+        **options : dict, optional
+            Additional sorting options
+            As allowed by :class:`SortOptions`
```

Comment:
Good idea to just pass through options, that automatically also allows to use `null_placement`.

I know this are indeed options, but for consistency I would suggest to use `**kwargs`. For example Array.sum also uses kwargs to pass through options.

Reply:
Another nit: I see we have this in other places as well, but the "dict" in the docstring as type for `**options` is quite misleading .. (you can't pass a dict in practice as argument)

(but since we use this everywhere, we can tackle that in a separate PR)

Reply:
Will rename to `kwargs`, regarding the `dict` type, that's what the numpydoc documentation suggests -> https://numpydoc.readthedocs.io/en/latest/format.html#parameters

> When documenting variable length positional, or keyword arguments, leave the leading star(s) in front of the name:
> ```
> *args : tuple
>     Additional arguments should be passed as keyword arguments
> **kwargs : dict, optional
>     Extra arguments to `metric`: refer to each metric documentation for a
>     list of all possible arguments.
> ```